### ISSUE TITLE
NaN KL divergence on Gemma-4 12b-it

### ISSUE THREAD

**Issue by Vinay-Umrethe** [p-e-w/heretic#346]
### Model

```bash
heretic --model google/gemma-4-12B-it --batch-size 128
```

```bash
Running trial 5 of 200...
* Parameters:
  * direction_index = per layer
  * attn.o_proj.max_weight = 1.21
  * attn.o_proj.max_weight_position = 30.99
  * attn.o_proj.min_weight = 0.33
  * attn.o_proj.min_weight_distance = 6.25
  * mlp.down_proj.max_weight = 1.32
  * mlp.down_proj.max_weight_position = 34.42
  * mlp.down_proj.min_weight = 1.18
  * mlp.down_proj.min_weight_distance = 22.99
* Resetting model...
* Abliterating...
* Evaluating...
  * Obtaining first-token probability distributions...
  * KL divergence: nan
  * Counting model refusals...
  * Refusals: 20/100
```

> [!TIP]
> the NEW gemma-4 12b does **not** use `Gemma4ForConditionalGeneration` which is used by E2B-it and E4B-it variants.
>
> It uses `Gemma4UnifiedForConditionalGeneration` as of transformers version `v5.10.1` latest

This one as well does not seem to be a Heretic issue, but `transformers` (modeling code / config.json might be incorrect)

**Comment by p-e-w** [comment]
That sounds like the logits returned by `generate` are already wrong, because I don't think the KLD can be NaN if both inputs are valid probability distributions.

**Comment by MoonRide303** [comment]
@p-e-w could you take a look at alternative solution in #350 (using raw generation logits for KL divergence)?

**Comment by p-e-w** [comment]
Fixed in #350.
